### PR TITLE
Make explicit db required for calls using Arrow

### PR DESCRIPTION
**Pull request by adamnsch**

### CODE REVIEW

File: graphdatascience/query_runner/arrow_query_runner.py [hunk 1]
```diff
@@ -81,18 +86,25 @@ def run_query_with_logging(self, query: str, params: Optional[Dict[str, Any]] =
...
+    def database(self) -> Optional[str]:
         return self._fallback_query_runner.database()
 
     def close(self) -> None:
         self._fallback_query_runner.close()
 
     def _run_arrow_property_get(self, graph_name: str, procedure_name: str, configuration: Dict[str, Any]) -> DataFrame:
+        if not self.database():
+            raise ValueError(
+                "For this call you must have explicitly specified a valid Neo4j database to target, "
```
Comment (s1ck) [inline graphdatascience/query_runner/arrow_query_runner.py:104]: maybe `... to execute on` instead of `.. to target`?

**Review by s1ck — APPROVED** [review]
Looks nice, just minor comments / suggestions.

File: graphdatascience/query_runner/arrow_query_runner.py (hunk 1 above)
Comment (adamnsch) [reply graphdatascience/query_runner/arrow_query_runner.py:104]: Sure!

File: README.md [hunk 2]
```diff
@@ -109,7 +109,7 @@ G = gds.alpha.graph.construct(
 )
 ```
 
-If your server uses GDS Enterprise edition and you have [enabled its Arrow Apache server](https://neo4j.com/docs/graph-data-science/current/installation/installation-apache-arrow/), the construction will be a lot faster.
+If your server uses GDS Enterprise edition and you have [enabled its Arrow Apache server](https://neo4j.com/docs/graph-data-science/current/installation/installation-apache-arrow/), the construction will be a lot faster. In this case you must also make sure that you have explicitly specified which database to use via `GraphDataScience.set_database`.
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply README.md:112]: Yes!

[the remark this answers is not in the document]
Comment (adamnsch) [reply README.md]: done

**Pull request merged**